sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricFeedbackLiveTests.cs

```diff
@@ -0,0 +1,290 @@
+            var periodValue = 10;
+
+            var feedbackToAdd = new MetricPeriodFeedback(MetricId, filter, PeriodType.AutoDetect, periodValue);
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Period));
+
+            var periodFeedback = addedFeedback as MetricPeriodFeedback;
+
+            Assert.That(periodFeedback, Is.Not.Null);
+            Assert.That(periodFeedback.PeriodType, Is.EqualTo(PeriodType.AutoDetect));
+            Assert.That(periodFeedback.PeriodValue, Is.EqualTo(periodValue));
+        }
+
+        [RecordedTest]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task GetAllFeedback(bool populateOptionalMembers)
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var options = new GetAllFeedbackOptions();
+
+            if (populateOptionalMembers)
+            {
+                options.TimeMode = FeedbackQueryTimeMode.FeedbackCreatedTime;
+                options.StartTime = FeedbackSamplingStartTime;
+                options.EndTime = FeedbackSamplingEndTime;
+                options.FeedbackType = FeedbackType.Comment;
+
+                options.Filter.AddDimensionColumn("city", "Delhi");
+            }
+
+            var feedbackCount = 0;
+
+            await foreach (MetricFeedback feedback in client.GetAllFeedbackAsync(MetricId, options))
+            {
+                Assert.That(feedback, Is.Not.Null);
+                Assert.That(feedback.Id, Is.Not.Null.And.Not.Empty);
+                Assert.That(feedback.MetricId, Is.EqualTo(MetricId));
+                Assert.That(feedback.UserPrincipal, Is.Not.Null.And.Not.Empty);
+                Assert.That(feedback.CreatedTime, Is.Not.Null);
+
+                Assert.That(feedback.DimensionFilter, Is.Not.Null);
+                Assert.That(feedback.DimensionFilter.DimensionFilter, Is.Not.Null);
+
+                ValidateGroupKey(feedback.DimensionFilter.DimensionFilter);
+
+                if (populateOptionalMembers)
+                {
+                    Assert.That(feedback.CreatedTime, Is.GreaterThanOrEqualTo(FeedbackSamplingStartTime));
+                    Assert.That(feedback.CreatedTime, Is.LessThanOrEqualTo(FeedbackSamplingEndTime));
+                    Assert.That(feedback.Type, Is.EqualTo(FeedbackType.Comment));
+
+                    Dictionary<string, string> dimensionColumns = feedback.DimensionFilter.DimensionFilter.AsDictionary();
+
+                    Assert.That(dimensionColumns.ContainsKey("city"));
+                    Assert.That(dimensionColumns["city"], Is.EqualTo("Delhi"));
+                }
+
+                if (feedback.Type == FeedbackType.Anomaly)
```

Comment:
should this be a switch instead?

Reply:
I'm personally not a fan of switches when there's complex logic inside each case. It's an extensible enum, though, so it's not supported.

Reply:
sad for our extensible enums :(